src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlBulkCopyColumnOrderHintCollectionTest.cs

```diff
@@ -0,0 +1,577 @@
+            Assert.Same(item1, collection[0]);
+            var item2 = new SqlBulkCopyColumnOrderHint("column2", SortOrder.Descending);
+            Assert.Same(item2, collection.Add(item2));
+            Assert.Same(item2, collection[1]);
+
+            IList list = CreateCollection();
+            int index = list.Add(item1);
+            Assert.Equal(0, index);
+            Assert.Same(item1, list[0]);
+            index = list.Add(item2);
+            Assert.Equal(1, index);
+            Assert.Same(item2, list[1]);
+        }
+
+        [Fact]
+        public void Add_HelperOverloads_ItemsAddedAsExpected()
+        {
+            SqlBulkCopyColumnOrderHintCollection collection = CreateCollection();
+            SqlBulkCopyColumnOrderHint item;
+
+            item = collection.Add("column1", SortOrder.Ascending);
+            Assert.NotNull(item);
+            Assert.Equal("column1", item.Column);
+            Assert.Equal(SortOrder.Ascending, item.SortOrder);
+        }
+
+        [Fact]
+        public void Add_InvalidItems_ThrowsArgumentException()
+        {
+            SqlBulkCopyColumnOrderHintCollection collection = CreateCollection();
+            Assert.Throws<ArgumentException>(() => collection.Add(new SqlBulkCopyColumnOrderHint(null, SortOrder.Ascending)));
+            Assert.Throws<ArgumentException>(() => collection.Add(new SqlBulkCopyColumnOrderHint(null, SortOrder.Unspecified)));
+            Assert.Throws<ArgumentException>(() => collection.Add(new SqlBulkCopyColumnOrderHint("", SortOrder.Descending)));
+            Assert.Throws<ArgumentException>(() => collection.Add(new SqlBulkCopyColumnOrderHint("", SortOrder.Unspecified)));
+            Assert.Throws<ArgumentException>(() => collection.Add(new SqlBulkCopyColumnOrderHint("column", SortOrder.Unspecified)));
+
+            IList list = CreateCollection();
+            Assert.Throws<ArgumentException>(() => list.Add(new SqlBulkCopyColumnOrderHint(null, SortOrder.Ascending)));
+            Assert.Throws<ArgumentException>(() => list.Add(new SqlBulkCopyColumnOrderHint(null, SortOrder.Unspecified)));
+            Assert.Throws<ArgumentException>(() => list.Add(new SqlBulkCopyColumnOrderHint("", SortOrder.Descending)));
+            Assert.Throws<ArgumentException>(() => list.Add(new SqlBulkCopyColumnOrderHint("", SortOrder.Unspecified)));
+            Assert.Throws<ArgumentException>(() => list.Add(new SqlBulkCopyColumnOrderHint("column", SortOrder.Unspecified)));
+        }
+
+        [Fact]
+        public void IListAddInsert_InsertNonSqlBulkCopyColumnOrderHintItems_DoNotThrow()
+        {
+            IList list = CreateCollection();
+            list.Add(new SqlBulkCopyColumnOrderHint("column", SortOrder.Descending));
+
+            // The following operations should really throw ArgumentException due to the
+            // mismatched types, but do not throw in the full framework.
+            string bogus = "Bogus";
```

Comment:
Can we use some other term here (foo/bar)? I wonder if SDL tools will catch this as offensive keyword.